### PR TITLE
Implementation of the card Clear the Land

### PR DESCRIPTION
**Pull request by davidmfritz**
My first card implementation, feedback appreciated!

### CODE REVIEW

File: Mage.Sets/src/mage/cards/c/ClearTheLand.java [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+    @Override
+    public boolean apply(Game game, Ability source) {
+        
+        int numOfCardsToReveal = 5;
+        boolean tapped = true;
+        
+        Player controller = game.getPlayer(source.getControllerId());
+        
+        if (controller != null) {
+            Players allPlayers = game.getPlayers();
```
Comment (JayDi85) [inline Mage.Sets/src/mage/cards/c/ClearTheLand.java]: You must use players in range, not all players:
`game.getState().getPlayersInRange(controller.getId(), game)`

File: Mage.Sets/src/mage/cards/c/ClearTheLand.java [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+                        }
+                        Card card = library.getFromTop(game);
+                        cardsToReveal.add(card);
+                        if (card.isLand()) {
+                            player.moveCards(card, Zone.BATTLEFIELD, source, game, tapped, false, true, null);
+                        } else {
+                            player.moveCards(card, Zone.EXILED, source, game);
+                        }
+                    }
+                    player.revealCards(source, "Revealed cards for " + player.getName(), cardsToReveal, game);
```
Comment (JayDi85) [inline Mage.Sets/src/mage/cards/c/ClearTheLand.java]: moves and exiles after reveal:
player.revealCards
player.moveCards battlefield
player.moveCards exile

**Comment by JayDi85** [comment]
Not yet... you must move all cards by one call too, not one by one... As example: players can select order of the moved cards to put it on battlefield/graveyard.

> Each player reveals the top five cards of his or her library, puts all land cards revealed this way onto the battlefield tapped, and exiles the rest.
 * Implement rules step by step:
 * > Each player
   * Cycle by players
 * > reveals the top five cards of his or her library
   * Collect 5 cards and reveal it all by one call.

 * > puts all land cards revealed this way onto the battlefield tapped
   * Find all lands from that cards and put it to battlefield by one call

 * > and exiles the rest
   * Exile all other cards by one call

**Comment by davidmfritz** [comment]
Thank you again for the feedback, I'm definitely learning something right here.
Could you please again review my changes? Thanks

**Comment by JayDi85** [comment]
I think that's fine. Thanks for contribution.

**Comment by davidmfritz** [comment]
Do I have to add a check if the cards to put onto the battlefield/into exile variable is empty before calling player.moveCards()?

**Comment by JayDi85** [comment]
> Do I have to add a check if the cards to put onto the battlefield/into exile variable is empty before calling player.moveCards()?

Nope, it's checks by inner functions too.

**Comment by davidmfritz** [comment]
Alright thank you again very much for the support!
(To be fair, I never seriously coded Java before ^^')

**Pull request merged**